tree/ntuple/v7/src/RNTupleDescriptor.cxx

```diff
@@ -872,18 +877,23 @@ ROOT::Experimental::RResult<void>
 ROOT::Experimental::Internal::RNTupleDescriptorBuilder::AddColumn(RColumnDescriptor &&columnDesc)
 {
    const auto fieldId = columnDesc.GetFieldId();
-   const auto index = columnDesc.GetIndex();
+   const auto columnIndex = columnDesc.GetIndex();
+   const auto representationIndex = columnDesc.GetRepresentationIndex();
 
    auto fieldExists = EnsureFieldExists(fieldId);
    if (!fieldExists)
       return R__FORWARD_ERROR(fieldExists);
-   if (fDescriptor.FindLogicalColumnId(fieldId, index) != kInvalidDescriptorId) {
+   if (fDescriptor.FindLogicalColumnId(fieldId, columnIndex, representationIndex) != kInvalidDescriptorId) {
       return R__FAIL("column index clash");
    }
-   if (index > 0) {
-      if (fDescriptor.FindLogicalColumnId(fieldId, index - 1) == kInvalidDescriptorId)
+   if (columnIndex > 0) {
+      if (fDescriptor.FindLogicalColumnId(fieldId, columnIndex - 1, representationIndex) == kInvalidDescriptorId)
          return R__FAIL("out of bounds column index");
    }
+   if (representationIndex > 0) {
+      if (fDescriptor.FindLogicalColumnId(fieldId, 0, representationIndex - 1) == kInvalidDescriptorId)
+         return R__FAIL("out of bounds representation index");
```

Comment:
Do we maybe want to check here that the previous column representations has all of its columns added? ie loop over `column = 0 .. GetColumnCardinality()`

Reply:
This and the next comment: I slightly patched the sanity checks. The result should be the same: when the descriptor is built, we ensure that all column representations have the same number of columns, and that the column index and the representation index are consecutive, starting at 0.